pkg/statestore/persistent/store.go

```diff
@@ -2,41 +2,42 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.package storage
 
-package statestore
+package persistent
```

Comment:
I would call this package leveldb, as there could be multiple persistent implementations.

Reply:
it will change to badger anyway at some point. i wouldn't like to couple the name with the underlying engine

Reply:
I think it should be underlying engine, unless we plan to have multiple implementations of persistent store inside this package.  Because, you have statestore, an one statestore could be inmem, and the other one leveldb, the third badger, etc.. I don't see a need for another general package like `persistent`, unless there is a need for another separation.

Reply:
For example, if you use database, you would usually have `database` interface, and concrete implementations, unless you need to split it on even more granularity such as sql, no-sql, etc.. But you wuold probably not have a package named nosql and then have a mongo implementation there.

Reply:
Plus, it would probably be nice to be able to have both leveldb and badger code implementations there at once, at least at some point, for test and benchmark, for example. Just my 2 c.

Reply:
done